controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +571,117 @@ func (r *OVNDBClusterReconciler) reconcileServices(
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Since DNS will always return ovsdbserver-xb.openstack.svc... this fields
+	// are an string instead of an array of strings. The add and removal of entries
+	// will be done using the DNSData CR, but this fields will remain the same on scale-up/down
+	dbAddress := ""
+
+	// DNSData is needed to provide DNS outside the cluster (edpm nodes) which needs to have
+	// spec.NetworkAttachment != ""
+	if instance.Spec.NetworkAttachment != "" {
+		for i, ovnPod := range podList.Items {
+			if i >= int(*(instance.Spec.Replicas)) {
+				Log.Info(fmt.Sprintf("In iteration %v is OOO, exit", i))
+				break
+			}
+			var dnsName string
+			var dnsIP string
+			var hostnames []string
+			dnsName = "dns-" + ovnPod.Name
+			// Get Hostname
+			svc, err := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+			hostnames = append(hostnames, hostname)
+
+			// Get IP
+			net_stat, err := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while getting the NAD for pod %s: %v", ovnPod.Name, err))
+				return ctrl.Result{}, err
+			}
+			for _, v := range net_stat {
+				if v.Interface == instance.Spec.NetworkAttachment {
+					// Return only IPv4 at the moment
+					if strings.Count(v.IPs[0], ":") == 0 {
+						dnsIP = v.IPs[0]
+						break
+					}
+				}
+			}
+
+			// Create DNSRecord
+			var DNSHosts []infranetworkv1.DNSHost
+			// ovsdbserver-xb-x entry
+			DNSHost := infranetworkv1.DNSHost{}
+			DNSHost.IP = dnsIP
+			DNSHost.Hostnames = hostnames
+			// ovsdbserver-xb entry
+			headless_dns_hostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
+			if instance.Spec.DBType == v1beta1.NBDBType {
+				headless_dns_hostname = ovndbcluster.ServiceNameNB + "." + instance.Namespace + ".svc"
+			}
+			dbAddress = fmt.Sprintf("tcp:%s:%d", headless_dns_hostname, svc.Spec.Ports[0].Port)
+			DNSHostCname := infranetworkv1.DNSHost{}
+			DNSHostCname.IP = dnsIP
+			DNSHostCname.Hostnames = append(DNSHostCname.Hostnames, headless_dns_hostname)
+			DNSHosts = append(DNSHosts, DNSHost)
+			DNSHosts = append(DNSHosts, DNSHostCname)
+
+			// Create DNSData
+			DNSData := &infranetworkv1.DNSData{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dnsName,
+					Namespace: ovnPod.Namespace,
+				},
+			}
+			_, err = controllerutil.CreateOrPatch(ctx, helper.GetClient(), DNSData, func() error {
+				DNSData.Spec.Hosts = DNSHosts
+				DNSData.Spec.DNSDataLabelSelectorValue = "dnsdata"
+				err := controllerutil.SetControllerReference(helper.GetBeforeObject(), DNSData, helper.GetScheme())
+				if err != nil {
+					return err
+				}
+				return nil
+			})
+			if err != nil {
+				Log.Error(err, "Error Creating DNSData.")
+			}
+
 		}
+
 	}
+
+	// Set DB Addresses
```

Comment:
the comment is redundant. Also... "addressES"? I see it's a single address.

Reply:
Done